internal/rows/arrowbased/batchloader_test.go

```diff
@@ -0,0 +1,89 @@
+package arrowbased
+
+import (
+	"bytes"
+	"context"
+	"github.com/apache/arrow/go/v11/arrow"
+	"github.com/apache/arrow/go/v11/arrow/array"
+	"github.com/apache/arrow/go/v11/arrow/ipc"
+	"github.com/apache/arrow/go/v11/arrow/memory"
+	"github.com/databricks/databricks-sql-go/internal/cli_service"
+	"github.com/databricks/databricks-sql-go/internal/config"
+	"github.com/pkg/errors"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"testing"
+)
+
+func generateMockArrowBytes() []byte {
+	mem := memory.NewCheckedAllocator(memory.NewGoAllocator())
+	defer mem.AssertSize(nil, 0)
+
+	fields := []arrow.Field{
+		{Name: "id", Type: arrow.PrimitiveTypes.Int32},
+		{Name: "name", Type: arrow.BinaryTypes.String},
+	}
+	schema := arrow.NewSchema(fields, nil)
+
+	builder := array.NewRecordBuilder(mem, schema)
+	defer builder.Release()
+
+	builder.Field(0).(*array.Int32Builder).AppendValues([]int32{1, 2, 3}, nil)
+	builder.Field(1).(*array.StringBuilder).AppendValues([]string{"one", "two", "three"}, nil)
+
+	record := builder.NewRecord()
+
+	var buf bytes.Buffer
+	w := ipc.NewWriter(&buf, ipc.WithSchema(record.Schema()))
+	if err := w.Write(record); err != nil {
+		return nil
+	}
+	if err := w.Close(); err != nil {
+		return nil
+	}
+	return buf.Bytes()
+}
+
+func TestBatchLoader(t *testing.T) {
+	testTable := []struct {
+		name             string
+		server           *httptest.Server
+		expectedResponse []*sparkArrowBatch
+		expectedErr      error
+	}{
+		{
+			name: "cloud-fetch-happy-case",
+			server: httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
If you have more than one test case isn't this going to be a problem? Since each test case is starting a server instance.

Reply:
Changed this in the latest commit - httptest server is now created before the test cases are run